cmake/FetchMongoC.cmake

```
@@ -2,43 +2,56 @@
-endif()
-
-# Declare mongo-c-driver as a dependency
-FetchContent_Declare(
-    mongo-c-driver
-    GIT_REPOSITORY https://github.com/mongodb/mongo-c-driver.git
-    GIT_TAG ${LIBMONGOC_DOWNLOAD_VERSION}
-
-    ${fetch_args}
-)
-
-FetchContent_GetProperties(mongo-c-driver)
-
-if(NOT mongo-c-driver_POPULATED)
-    set(OLD_ENABLE_TESTS ${ENABLE_TESTS})
-    set(OLD_BUILD_TESTING ${BUILD_TESTING})
-    set(OLD_CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS})
-    set(OLD_CMAKE_C_FLAGS ${CMAKE_C_FLAGS})
-
-    # Set ENABLE_TESTS to OFF to disable the test-libmongoc target in the C driver.
-    # This prevents the LoadTests.cmake script from attempting to execute test-libmongoc.
-    # test-libmongoc is not built with the "all" target.
-    # Attempting to execute test-libmongoc results in an error: "Unable to find executable: NOT_FOUND"
-    set(ENABLE_TESTS OFF)
-    set(BUILD_TESTING OFF)
-    string(REPLACE " -Werror" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
-    string(REPLACE " -Werror" "" CMAKE_C_FLAGS "${CMAKE_C_FLAGS}")
-    FetchContent_MakeAvailable(mongo-c-driver)
-    set(CMAKE_CXX_FLAGS ${OLD_CMAKE_CXX_FLAGS})
-    set(CMAKE_C_FLAGS ${OLD_CMAKE_C_FLAGS})
-    set(ENABLE_TESTS ${OLD_ENABLE_TESTS})
-    set(BUILD_TESTING ${OLD_BUILD_TESTING})
-endif()
-
-message(STATUS "Download and configure C driver version ${LIBMONGOC_DOWNLOAD_VERSION} ... end")
+function(fetch_mongoc)
+    message(STATUS "Download and configure C driver version ${LIBMONGOC_DOWNLOAD_VERSION} ... end")
+
+    set(fetch_args "")
+    if (CMAKE_VERSION VERSION_GREATER_EQUAL "3.25.0")
+        list(APPEND fetch_args "SYSTEM")
+    endif()
+
+    # Declare mongo-c-driver as a dependency
+    FetchContent_Declare(
+        mongo-c-driver
+        GIT_REPOSITORY https://github.com/mongodb/mongo-c-driver.git
+        GIT_TAG ${MONGOC_DOWNLOAD_VERSION}
+
+        ${fetch_args}
+    )
+
+    FetchContent_GetProperties(mongo-c-driver)
+
+    if(NOT mongo-c-driver_POPULATED)
+        # Must ensure BUILD_VERSION is not inherited either as a normal variable or as a cache variable.
+        unset(OLD_BUILD_VERSION)
+        if(DEFINED BUILD_VERSION)
+            set(OLD_BUILD_VERSION ${BUILD_VERSION})
+            unset(BUILD_VERSION)
+        endif()
+        unset(OLD_CACHE_BUILD_VERSION)
+        if(DEFINED CACHE{BUILD_VERSION})
+            set(OLD_CACHE_BUILD_VERSION $CACHE{BUILD_VERSION})
+            unset(BUILD_VERSION CACHE)
+        endif()
+
+        # Disable unnecessary targets and potential conflicts with C++ Driver options.
+        set(ENABLE_TESTS OFF)
+        set(BUILD_TESTING OFF)
+        string(REPLACE " -Werror" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+        string(REPLACE " -Werror" "" CMAKE_C_FLAGS "${CMAKE_C_FLAGS}")
+
+        FetchContent_MakeAvailable(mongo-c-driver)
+
+        # Restore prior value of BUILD_VERSION only when they were previously set.
+        if(DEFINED OLD_BUILD_VERSION)
+            set(BUILD_VERSION ${OLD_BUILD_VERSION})
+        endif()
```

```suggestion
        # Restore prior value of cached BUILD_VERSION only when previously set.
```

IIUC restoring the normal variable is not needed, since the `unset(BUILD_VERSION)` will not impact the parent scope.